src/shared/ethtool-util.c

```diff
@@ -1101,3 +1102,169 @@ int config_parse_wol(
 
         return 0;
 }
+
+int config_parse_nic_coalesce_setting(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata
+) {
+        u32_opt *dst = data;
+        uint32_t k;
+        int r;
+
+        if (isempty(rvalue)) {
+                dst->value = 0;
+                dst->set = false;
+                return 0;
+        }
+
+        r = safe_atou32(rvalue, &k);
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Failed to parse coalesce setting value, ignoring: %s", rvalue);
+                return 0;
+        }
+
+        if (STR_IN_SET(lvalue, "StatsBlockCoalesceUSec", "CoalesceRateSampleInterval") && k < 1) {
+                log_syntax(unit, LOG_WARNING, filename, line, 0,
+                           "Invalid %s= value, ignoring: %s", lvalue, rvalue);
+                return 0;
+        }
```

Comment:
please use separate parser functions if the fields are sufficeintly different. also use parse_sec() for parsing time values.

StatsBlockCoalesceUSec should probably be called StatisticsBlockCoalesceSec=. i.e, please no unnecessary abbreviations. And we use seconds as default time unit everywhere, please do so here too. If people specify other time units explicitly that takes precedence of course, and parse_sec() takes care of that.